src/welcome.rs

```diff
@@ -126,6 +128,90 @@ pub(crate) fn assign_talk_role(cx: &Context, reaction: &Reaction) -> Result<(),
     Ok(())
 }
 
+pub(crate) fn update_welcome_message(args: Args) -> Result<(), Error> {
+    use std::str::FromStr;
+
+    let new_message = &args
+        .params
+        .get("message")
+        .ok_or("unable to retrieve message param")?;
+
+    let welcome_message_is_not_cached = {
+        let data = args.cx.data.read();
```

Comment:
Say this function was ran twice in parallel. This check could be hit twice before the message is stored in `data`, which could cause parallel attempts to populate it.

Reply:
You are absolutely correct, nice catch, this is a problem and I will address it.